editoast/src/views/timetable/stdcm.rs

```diff
@@ -213,6 +221,23 @@ async fn stdcm(
         })
         .await?;
 
+    let towed_rolling_stock =
+        if let Some(towed_rolling_stock_id) = stdcm_request.towed_rolling_stock_id {
+            let towed_rolling_stock =
+                TowedRollingStockModel::retrieve_or_fail(conn, towed_rolling_stock_id, || {
+                    STDCMError::TowedRollingStockNotFound {
+                        towed_rolling_stock_id,
+                    }
+                })
+                .await?;
+            Some(towed_rolling_stock)
+        } else {
+            None
+        };
```

Comment:
I believe `map` and [`transpose`](https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose) can be used here.

Reply:
I tried, how you deal with an async call inside the map ?

Reply:
I missed the `await`... good point, resolving 👀